fenix/app/src/test/java/org/mozilla/fenix/shopping/store/ReviewQualityCheckStoreTest.kt

```diff
@@ -746,7 +747,7 @@ class ReviewQualityCheckStoreTest {
 
             val expected = ReviewQualityCheckState.OptedIn(
                 productReviewState = ProductAnalysisTestData.analysisPresent(
-                    analysisStatus = AnalysisStatus.REANALYZING,
+                    analysisStatus = AnalysisStatus.Reanalyzing(0.5f),
```

Comment:
Updated existing tests, but not totally sure what other tests should be added

Reply:
Ideally, we want to test that the progress returned from `ReviewQualityCheckService` is being update to state. We have the fake for that already, now we need to make sure that `analysisStatus()` returns multiple times with responses from { "in_progress", 0.0..100.0 } and eventually { "completed", 100.0} and assert that the state should be "reanalyzing" with the respective normalized float values (0 - 1.0f).

Reply:
Updated this test to dispatch multiple `UpdateAnalysisProgress` and check the state for corresponding progress values. Also tried to add some `noAnalysisPresent` tests but had some issues with the expected value.